sql/triggers.sql

```diff
@@ -0,0 +1,37 @@
+DELIMITER //
+
+DROP TRIGGER IF EXISTS specimen_weight_update //
+
+CREATE TRIGGER specimen_weight_update
+AFTER INSERT ON Aliquot
+FOR EACH ROW
+BEGIN
+    -- Don't fire the trigger when this variable is disabled
+    IF (@DISABLE_TRIGGERS = FALSE)
+    THEN
+        -- Get the current weight of the associated Specimen
+        -- NEW is a temporary table containing the data from the INSERT call
+        SELECT SpecimenWeight INTO @CurrentSpecimenWeight FROM Specimen WHERE idSpecimen = NEW.Specimen_idSpecimen;
+        -- Subject the weight of the new Aliquot from the Specimen
+        UPDATE Specimen SET SpecimenWeight = @CurrentSpecimenWeight - NEW.AliquotWeight WHERE idSpecimen = NEW.Specimen_idSpecimen;
+    END IF;
+END //
+
+
+DROP TRIGGER IF EXISTS aliquot_weight_update //
+
+CREATE TRIGGER aliquot_weight_update
+AFTER INSERT ON Sample
+FOR EACH ROW
+BEGIN
+    IF (@DISABLE_TRIGGERS = FALSE)
```

Comment:
why not @ENABLE_TRIGGERS = TRUE ?
doesn't really matter much, I think it reads a little better

Reply:
Because the variable is part of the user's session. If the variable isn't set yet, it doesn't exist and the trigger will fire. So it is disabling the trigger not enabling it.

Reply:
Really tho it should be disable_triggers = true

Reply:
makes sense, might make for a good comment

Reply:
I guess now that I think aobut it, seems self evident enough